src/main/java/com/marklogic/kafka/connect/source/XmlRowBatcherBuilder.java

```diff
@@ -0,0 +1,89 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.RowBatchSuccessListener;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.io.DOMHandle;
+import com.marklogic.client.io.marker.ContentHandle;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Map;
+
+public class XmlRowBatcherBuilder extends AbstractRowBatchBuilder implements RowBatcherBuilder<Document> {
+    private static final String TABLE_NS_URI = "http://marklogic.com/table";
+    private final DocumentBuilderFactory documentBuilderFactory;
+
+    XmlRowBatcherBuilder(DataMovementManager dataMovementManager, Map<String, Object> parsedConfig) {
+        super(dataMovementManager, parsedConfig);
+        documentBuilderFactory = DocumentBuilderFactory.newInstance();
+    }
+
+    public RowBatcher<Document> newRowBatcher(List<SourceRecord> newSourceRecords) {
+        ContentHandle<Document> domHandle = new DOMHandle().withMimetype("application/xml");
+        RowBatcher<Document> rowBatcher =  dataMovementManager.newRowBatcher(domHandle);
+        configureRowBatcher(parsedConfig, rowBatcher);
+        rowBatcher.onSuccess(event -> onSuccessHandler(event, newSourceRecords));
+        return rowBatcher;
+    }
+
+    protected void onSuccessHandler(RowBatchSuccessListener.RowBatchResponseEvent<Document> event, List<SourceRecord> newSourceRecords) {
+        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        Transformer transformer;
+        try {
+            transformer = transformerFactory.newTransformer();
+            Document rowsDoc = event.getRowsDoc();
+
+            Element docElement = rowsDoc.getDocumentElement();
+            Node columnsNode = docElement.getElementsByTagNameNS(TABLE_NS_URI, "columns").item(0);
+            NodeList rows = docElement.getElementsByTagNameNS(TABLE_NS_URI, "row");
+
+            for (int i=0; i < rows.getLength(); i++) {
+                Document newDocument = convertRowNodeToNewDocument(columnsNode, rows.item(i), transformer);
```

Comment:
Sorry, but I now realize we need two try/catch's for XML handling - 1) the overall one, to catch any JAXP issues, or possibly issues with obtaining the "row" and "columnsNode" nodes; and 2) a try/catch for every row in the `for` loop, similar to what's done for CSV/JSON. CSV/JSON don't need that first try/catch because we don't have any meaningful initialization to do first. 

Alternatively, you could move the code for doing the JAXP and XML parsing stuff into a private method that returns a Tuple<Node, Node> - the first Node being the columns, the second Node being the rows. And you could wrap that in a try/catch so that if we failed to get those two nodes, we log the error and don't do anything further. If we do get those nodes, then we hop into the for loop and do a try/catch on each row.